File: Web_demo/frontend/src/types/news.ts
// src/types/news.ts

export interface CrawledNews {
  title: string;
  body: string;
  source: string;
  url?: string | null;
  published_at?: string | null;

  summary: string;
  category: string;  // Category từ URL
}

export interface PreviewRequest {
  title?: string;
  body: string;
}

export interface PreviewResponse {
  summary: string;

  category: string;
  score: number;

  secondary_category?: string | null;
  secondary_score?: number | null;
}

File: Web_demo/backend/app/schemas/news.py
#app\schemas\news.py
from pydantic import BaseModel
from typing import Optional, List


class CrawlRequest(BaseModel):
    
    sources: List[str] = ["vnexpress", "vietnamnet"]
    limit: int = 20
    force_new: bool = False
    force_refresh: bool = False  # Bắt buộc chạy model lại, không dùng cache


class CrawledNews(BaseModel):
    title: str
    body: str
    source: str
    url: Optional[str] = None
    published_at: Optional[str] = None

    summary: str
    category: str  # Category từ URL, không còn dùng model phân loại
